### PR TITLE
Use timescaledb prefix for compression errors

### LINKED ISSUE #1642: Suggestion: Improve error message when timescaledb.compress option is not specified
Running the command: "`ALTER TABLE <table_name> SET (timescale.compress,`..." as specified in the documentation results in the error message:

> `must set the 'compress' boolean option when setting compression options`

I suggest changing this error message (and any others similar) to say:

> `must set the 'timescaledb.compress' boolean option when setting compression options`

### PR DESCRIPTION
**Pull request by gayyappan**
Modify compression parameter related error messages
to make them consistent.
Fixes #1642

### CODE REVIEW

File: tsl/src/compression/create.c [hunk 1]
```diff
@@ -245,7 +245,7 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
 		{
 			ereport(ERROR,
 					(errcode(ERRCODE_SYNTAX_ERROR),
-					 errmsg("column %s in compress_segmentby list does not exist",
+					 errmsg("column %s in timescaledb.compress_segmentby list does not exist",
```
Comment (erimatnor) [inline tsl/src/compression/create.c]:
```suggestion
					 errmsg("column \"%s\" specified in option timescaledb.compress_segmentby does not exist",
```

I believe it is standard practice to double quote identifiers like columns in PostgreSQL error messages. 

Here's an example from PostgreSQL:

`"column \"%s\" of relation \"%s\" does not exist"`

File: tsl/src/compression/create.c [hunk 2]
```diff
@@ -261,16 +261,16 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
 		{
 			ereport(ERROR,
 					(errcode(ERRCODE_SYNTAX_ERROR),
-					 errmsg("column %s in compress_orderby list does not exist",
+					 errmsg("column %s in timescaledb.compress_orderby list does not exist",
```
Comment (erimatnor) [inline tsl/src/compression/create.c]:
```suggestion
					 errmsg("column \"%s\" in option timescaledb.compress_orderby does not exist",
```

File: tsl/src/compression/create.c [hunk 3]
```diff
@@ -261,16 +261,16 @@ compresscolinfo_init(CompressColInfo *cc, Oid srctbl_relid, List *segmentby_cols
...
 							NameStr(col->colname))));
 		}
 		/* check if orderby_cols and segmentby_cols are distinct */
 		if (segorder_colindex[col_attno - 1] != 0)
 		{
 			ereport(ERROR,
 					(errcode(ERRCODE_SYNTAX_ERROR),
-					 errmsg("cannot use the same column %s in compress_orderby and "
-							"compress_segmentby",
+					 errmsg("cannot use the same column %s in timescaledb.compress_orderby and "
```
Comment (erimatnor) [inline tsl/src/compression/create.c]:
```suggestion
					 errmsg("cannot use column \"%s\" in both timescaledb.compress_orderby and "
```

File: tsl/src/compression/create.c [hunk 4]
```diff
@@ -766,7 +766,8 @@ validate_existing_constraints(Hypertable *ht, CompressColInfo *colinfo)
 					{
 						ereport(ERROR,
 								(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-								 errmsg("constraint %s requires column %s to be a segment_by "
+								 errmsg("constraint %s requires column %s to be a "
```
Comment (erimatnor) [inline tsl/src/compression/create.c]:
```suggestion
								 errmsg("constraint \"%s\" requires column \"%s\" to be a "
```

File: tsl/src/compression/create.c [hunk 5]
```diff
@@ -786,11 +787,13 @@ validate_existing_constraints(Hypertable *ht, CompressColInfo *colinfo)
 					if (col_def->segmentby_column_index < 1 && col_def->orderby_column_index < 1)
 						ereport(ERROR,
 								(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-								 errmsg("constraint %s requires column %s to be a segment_by or "
-										"order_by column for compression",
+								 errmsg("constraint %s requires column %s to be a "
```
Comment (erimatnor) [inline tsl/src/compression/create.c]:
```suggestion
								 errmsg("constraint \"%s\" requires column \"%s\" to be a "
```

File: src/process_utility.c [hunk 6]
```diff
@@ -2896,7 +2896,8 @@ process_altertable_set_options(AlterTableCmd *cmd, Hypertable *ht)
 		if (parse_results[CompressEnabled].is_default)
 			ereport(ERROR,
 					(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
-					 errmsg("must set the 'compress' boolean option when setting compression "
+					 errmsg("must set the 'timescaledb.compress' boolean option when setting "
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply src/process_utility.c]: looks like we don't use the single quote in other places. So removing them and modified error message. Usage of timescaledb.compress is mandatory. The other parameters like compress_orderby, segmentby are optional.  (That's how it was spec-ed out partly to make sure that the options are not used erroneously. It is also consistent with continuous aggregate specification where usage of timescaledb.continuous_agg is mandatory before specifying any of the other parameters).

File: src/compression_with_clause.c [hunk 7]
```diff
@@ -50,8 +50,9 @@ throw_segment_by_error(char *segment_by)
 {
 	ereport(ERROR,
 			(errcode(ERRCODE_SYNTAX_ERROR),
-			 errmsg("unable to parse the compress_segmentby option '%s'", segment_by),
-			 errhint("The compress_segmentby option should be a comma separated list of column "
+			 errmsg("unable to parse the timescaledb.compress_segmentby option '%s'", segment_by),
```
Comment (svenklemm) [inline src/compression_with_clause.c]: hmm i think the "the" should be dropped here

File: src/compression_with_clause.c [hunk 8]
```diff
@@ -155,8 +156,9 @@ throw_order_by_error(char *order_by)
 {
 	ereport(ERROR,
 			(errcode(ERRCODE_SYNTAX_ERROR),
-			 errmsg("unable to parse the compress_orderby option '%s'", order_by),
-			 errhint("The compress_orderby option should be a comma separated list of column "
+			 errmsg("unable to parse the timescaledb.compress_orderby option '%s'", order_by),
```
Comment (svenklemm) [inline src/compression_with_clause.c]: drop the "the"

File: src/compression_with_clause.c (hunk 7 above)
Comment (gayyappan) [reply src/compression_with_clause.c]: fixed

**Pull request merged**